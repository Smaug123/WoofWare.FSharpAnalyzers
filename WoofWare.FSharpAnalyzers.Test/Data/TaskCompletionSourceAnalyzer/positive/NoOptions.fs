module TaskCompletionSourceNoOptions

open System.Threading.Tasks

let createTcsNoArgs () =
    let tcs = TaskCompletionSource<int> ()
    tcs.SetResult 42
    tcs.Task

let createTcsWithState () =
    let tcs = TaskCompletionSource<string> "state"
    tcs.SetResult "hello"
    tcs.Task

let createTcsMultiple () =
    let tcs1 = TaskCompletionSource<int> ()
    let tcs2 = TaskCompletionSource<string> ()
    tcs1.SetResult 1
    tcs2.SetResult "world"
    (tcs1.Task, tcs2.Task)
