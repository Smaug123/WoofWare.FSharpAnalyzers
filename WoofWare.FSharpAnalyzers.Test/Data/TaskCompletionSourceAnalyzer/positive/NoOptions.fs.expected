WOOF-TCS-ASYNC | Warning | (6,14--6,42) | TaskCompletionSource<T> created without TaskCreationOptions.RunContinuationsAsynchronously. This can cause continuations to run inline on the calling thread, leading to deadlocks, thread-pool starvation, and corruption of state. Always use: new TaskCompletionSource<T>(TaskCreationOptions.RunContinuationsAsynchronously). Suppress with comment including text 'TaskCompletionSource without RunContinuationsAsynchronously allowed'.
WOOF-TCS-ASYNC | Warning | (11,14--11,50) | TaskCompletionSource<T> created without TaskCreationOptions.RunContinuationsAsynchronously. This can cause continuations to run inline on the calling thread, leading to deadlocks, thread-pool starvation, and corruption of state. Always use: new TaskCompletionSource<T>(TaskCreationOptions.RunContinuationsAsynchronously). Suppress with comment including text 'TaskCompletionSource without RunContinuationsAsynchronously allowed'.
WOOF-TCS-ASYNC | Warning | (16,15--16,43) | TaskCompletionSource<T> created without TaskCreationOptions.RunContinuationsAsynchronously. This can cause continuations to run inline on the calling thread, leading to deadlocks, thread-pool starvation, and corruption of state. Always use: new TaskCompletionSource<T>(TaskCreationOptions.RunContinuationsAsynchronously). Suppress with comment including text 'TaskCompletionSource without RunContinuationsAsynchronously allowed'.
WOOF-TCS-ASYNC | Warning | (17,15--17,46) | TaskCompletionSource<T> created without TaskCreationOptions.RunContinuationsAsynchronously. This can cause continuations to run inline on the calling thread, leading to deadlocks, thread-pool starvation, and corruption of state. Always use: new TaskCompletionSource<T>(TaskCreationOptions.RunContinuationsAsynchronously). Suppress with comment including text 'TaskCompletionSource without RunContinuationsAsynchronously allowed'.
